app/src/main/java/com/github/se/polyfit/model/nutritionalInformation/NutritionalInformation.kt

```diff
@@ -53,10 +54,12 @@ class NutritionalInformation {
             val index = nutrients.indexOfFirst { it.nutrientType == newValues.nutrientType }
 
             nutrients[index] = (nutrients[index] + newValues)!!
-          } else {
+          } else if (newValues.amount >= 0.0) {
             nutrients.add(newValues.deepCopy())
           }
         }
+
+    nutrients.removeIf { it.amount < 0.0 }
```

Comment:
Do you think it would be a good idea to Log.d the case when this condition occurs? This way, we can track it in case a bug arises elsewhere. Or do you think that it is an necessary ?

Reply:
I think that's a good idea :-) Will add that in